eos-update-server/eos-update-server.c

```diff
@@ -492,25 +516,48 @@ main (int argc, char **argv)
       return EXIT_DISABLED;
     }
 
-  repo = eos_updater_local_repo ();
-  server = eos_updater_repo_server_new (repo,
-                                        options.served_remote,
-                                        NULL,
-                                        &error);
-  if (server == NULL)
+  /* Set up the server and repositories. */
+  soup_server = soup_server_new (NULL, NULL);
```

Comment:
Meh, ok. The annotation in the libsoup is likely wrong (there is no such thing for g_object_new, which looks similar).

Thanks for checking!